src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList(int size) {
+        if (size > 0) {
+            this.elementData = new Object[size];
+            this.size = 0;
+        } else if (size == 0) {
+            this.elementData = EMPTY_ELEMENTDATA;
+            this.size = 0;
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: " + size);
+        }
+    }
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= elementData.length) {
+            grow();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index " + index + "size " + size);
+        }
```

Comment:
I think we should also grow array here if it is full

Reply:
i've read that and implemented exactly so
![image](https://user-images.githubusercontent.com/85991807/163981843-45e4298d-63bc-4f00-ba17-4ab1662e226c.png)